### PR TITLE
v1.2a Add mod command

### PR DESCRIPTION
**Pull request by jonasgwt**
Add the `ModCommandParser` to parse all `mod` commands.
Add an abstract class `ModCommand`.
Add a class `ModAddCommand` that extends `ModCommand`.
Modify the `add` command to also accept mods with the prefix `m/`.
Modify the `edit` command to throw an exception when a user attempts to edit the mods.

Add tests for `ModAddCommand`
Add tests for `ModCommandParser`.
Update tests for `EditCommand`.
Update tests for `AddCommandParser`.
Update tests for `EditCommandParser`.
Update tests for `ParserUtil`.
Add tests for `Mod`.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ModCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     *
+     * @param args The user command.
+     * @return The index of the person in String.
+     */
+    private String getIndexFromCommand(String args) throws ParseException {
+        String[] splittedArgs = args.split(" ");
+        String index = splittedArgs[0];
+        final Matcher matcher = INDEX_FORMAT.matcher(index.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(ModCommand.MESSAGE_INDEX_EMPTY);
```
Comment (carriezhengjr) [inline src/main/java/seedu/address/logic/parser/ModCommandParser.java:109]: When I enter `mod add -1 hi` which contains a negative index, "Index cannot be empty!" is shown but the index is not really empty.

For other commands like `mod add 10 cs2100` (invalid index) and `mod add 1` (empty mod), the displayed messages are good. 👍 

Perhaps we can do it like the `edit` command, i.e. for other situations not covered by invalid index (exceeds range), empty index or empty mod, show a general error message.
e.g. `edit -1`, `edit -1 n/ca` triggers the following message:
```
Invalid command format! 
edit: Edits the details of the person identified by the index number used in the displayed person list. Existing values will be overwritten by the input values.
Parameters: INDEX (must be a positive integer) [n/NAME] [t/TELEGRAM] [g/GITHUB] [p/PHONE] [e/EMAIL] [tag/TAG]...
Example: edit 1 g/john_doe p/91234567 e/johndoe@example.com
```

**Review by carriezhengjr — COMMENTED** [review]
`mod add` function is working well. 👍  

Just some error messages that could be changed.

**Review by sltsheryl — APPROVED** [review]
Merge after making the above changes so I can get started on Javafx 🙂

**Comment by jonasgwt** [comment]
> `mod add` function is working well. 👍
> 
> Just some error messages that could be changed.

Okay thanks, I made the changes.

**Pull request merged**